src/main/src/main/java/org/geoserver/catalog/util/ReaderDimensionsAccessor.java

```diff
@@ -246,47 +272,92 @@ public Date getMaxTime() throws IOException {
+                return getTimeFormat().parse(fixedValue);
+            } catch (ParseException e) {
+                throw new RuntimeException("Failed to get minimum time from coverage reader", e);
+            }
+        } else {
+            final String currentTime =
+                    reader.getMetadataValue(AbstractGridCoverage2DReader.TIME_DOMAIN_MAXIMUM);
+            if (currentTime == null) {
+                return null;
+            }
+            try {
+                return getTimeFormat().parse(currentTime);
+            } catch (ParseException e) {
+                throw new RuntimeException("Failed to get CURRENT time from coverage reader", e);
+            }
+
         }
+
+
     }
 
-    /** Returns the min value for the time */
+    /**
+     * Returns the min value for the time
+     */
     public Date getMinTime() throws IOException {
         if (!hasTime()) {
             return null;
         }
-        final String currentTime =
-                reader.getMetadataValue(AbstractGridCoverage2DReader.TIME_DOMAIN_MINIMUM);
-        if (currentTime == null) {
-            return null;
-        }
-        try {
-            return getTimeFormat().parse(currentTime);
-        } catch (ParseException e) {
-            throw new RuntimeException("Failed to get minimum time from coverage reader", e);
+
+        if (dimensionInfo != null && dimensionInfo.getFixedValues() != null) {
+            String fixedValue = String.valueOf(getFixedValueRange(dimensionInfo.getFixedValues()));
+            try {
+                return getTimeFormat().parse(fixedValue);
+            } catch (ParseException e) {
+                throw new RuntimeException("Failed to get minimum time from coverage reader", e);
+            }
+        } else {
+            final String currentTime =
+                    reader.getMetadataValue(AbstractGridCoverage2DReader.TIME_DOMAIN_MINIMUM);
+            if (currentTime == null) {
+                return null;
+            }
+            try {
+                return getTimeFormat().parse(currentTime);
+            } catch (ParseException e) {
+                throw new RuntimeException("Failed to get minimum time from coverage reader", e);
+            }
         }
+
+
     }
 
-    /** Returns a {@link SimpleDateFormat} using the UTC_PATTERN and the UTC time zone */
+    /**
+     * Returns a {@link SimpleDateFormat} using the UTC_PATTERN and the UTC time zone
+     */
     public SimpleDateFormat getTimeFormat() {
         final SimpleDateFormat df = new SimpleDateFormat(UTC_PATTERN);
         df.setTimeZone(UTC_TIME_ZONE);
         return df;
     }
 
-    /** True if the reader has a elevation dimension */
+    /**
+     * True if the reader has a elevation dimension
+     */
     public boolean hasElevation() throws IOException {
         return "true".equalsIgnoreCase(reader.getMetadataValue(HAS_ELEVATION_DOMAIN));
     }
 
+    public TreeSet<Object> getFixedValueRange(String fixedValues) {
```

Comment:
If the purpose of this method is to return a parsed list of dimension values then the method should also parse the values. If the purpose of the method is to simply return a Collection of some kind with the different fixed value retrieved from the splitted string than I would name it asCollection(String fixedValues) or asList(String fixedValues).
More over this method is called in many different context that is not clear what you aim at when you call it. Can you clarify a bit?

Reply:
This method is still around and is unclear why it parse an array and add the string to a TreeSet. Can you either clarify or modify the code?

Reply:
This method is still around and is unclear why it parse an array and add the string to a TreeSet. Can you either clarify or modify the code?